pkg/chunk/cache/fifo_cache_test.go

```diff
@@ -2,100 +2,161 @@ package cache
 
 import (
 	"context"
-	"strconv"
+	"fmt"
 	"testing"
 	"time"
-	"unsafe"
 
 	"github.com/prometheus/client_golang/prometheus/testutil"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-const size = 10
-const overwrite = 5
+func TestFifoCacheEviction(t *testing.T) {
+	const (
+		cnt       = 10
+		overwrite = 5
+	)
+	itemTemplate := &cacheEntry{
+		key:   "00",
+		value: 0,
+	}
+	testFifoCacheEviction(t, "test-memory-eviction", FifoCacheConfig{MaxSizeBytes: cnt * sizeOf(itemTemplate), Validity: 1 * time.Minute})
+	testFifoCacheEviction(t, "test-items-eviction", FifoCacheConfig{MaxSizeItems: cnt, Validity: 1 * time.Minute})
+}
 
-func TestFifoCache(t *testing.T) {
-	c := NewFifoCache("test1", FifoCacheConfig{Size: size, Validity: 1 * time.Minute})
+func testFifoCacheEviction(t *testing.T, testname string, cfg FifoCacheConfig) {
+	const (
+		cnt       = 10
+		overwrite = 5
```

Comment:
The name `overwrite` makes thinking it's the number of keys which get `put()` again, while it's actually the number of evictions we want to trigger.